odxtools/diaglayers/diaglayer.py

```diff
@@ -124,7 +124,7 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
             # imported references only apply within this specific
             # diagnostic layer
             extended_odxlinks = copy(odxlinks)
-            extended_odxlinks.update(imported_links)
+            extended_odxlinks.update(imported_links, False)
```

Comment:
@andlaus this is the case that we did manage to catch, not sure if there are other cases that we missed

Reply:
as far as I'm aware, IMPORT-REFs should be the only place were this is needed. on the other hand, this could probably also be implemented by reversing the logic, i.e. copying the imported links and overriding them with the local ones? i.e., something like
```python
            extended_odxlinks = copy(imported_links)
            extended_odxlinks.update(odxlinks)
```

(maybe I'm missing something, though.)

Reply:
come think about it, we can probably rename the `imported_links` variable to `extended_links` and replace these two lines by `extended_odxlinks.update(odxlinks)`?

Reply:
We tried that, `copy(imported_links)` crashed, don't recall the error now

Reply:
we can't because `odxlinks` is an `OdxLinkDatabase` object while `imported_links` is a dictionary. (I think the approach outlined by the PR is the best way forward.)